Update installation instructio for JAX and add some dependencies. (#117)
NVIDIA/TransformerEngine: README.rst

From source
^^^^^^^^^^^

> For JAX, you need to install those dependencies first. (We are working to make this automatic):

Reviewer: I don't think we need to say that we work to make this automatic - at some point we will just drop this section :-).

From source
^^^^^^^^^^^

> For JAX and Tensorflow, you need to install that dependency first:

Reviewer 2: ```suggestion
For JAX and Tensorflow, pybind11 must be installed:
```

For JAX and Tensorflow, you need to install that dependency first:

.. code-block:: bash

>   apt-get install ninja-build pybind11-dev

Reviewer: Why is pybind11-dev package not working?

Author: It is working. It is just that pip packges works for more OS.
So this is a more general solution.
rpm based OS will work with the pip package.

> Then, you can install this optional dependency:

Reviewer 2: ```suggestion
Then, the following optional dependencies can be installed:
```

Reviewer 2: Stating the purpose of these dependencies would be useful, since they are 'optional'.

.. code-block:: bash

  pip install ninja

> Clone the repository and inside it type one of those commands:

Reviewer 2: ```suggestion
Install TE (optionally specifying the framework):
```

Author: I added the clone command line.

Reviewer: Then "type one of those commands" is even more ambiguous. I agree with Kithi's suggestion.

.. code-block:: bash

>   NVTE_FRAMEWORK=all pip install .        # Building with all frameworks.
>   NVTE_FRAMEWORK=pytorch pip install .    # Building with pyTorch only.
>   NVTE_FRAMEWORK=jax pip install .        # Building with JAX only.
>   NVTE_FRAMEWORK=tensorflow pip install . # Building with Tensorflow only.

Reviewer 2: ```suggestion
  NVTE_FRAMEWORK=all pip install .        # Build TE for all supported frameworks.
  NVTE_FRAMEWORK=pytorch pip install .    # Build TE for PyTorch only.
  NVTE_FRAMEWORK=jax pip install .        # Build TE for JAX only.
  NVTE_FRAMEWORK=tensorflow pip install . # Build TE for TensorFlow only.
```

> The default is pyTorch only.

Reviewer 2: ```suggestion
If the framework is not explicitly specified, TE will be built for PyTorch only.
```
